doc/nrf/app_dev/bootloaders_dfu/dfu_tools_mcumgr_cli.rst

```diff
@@ -0,0 +1,521 @@
+.. _dfu_tools_mcumgr_cli:
+
+Command-line tool
+#################
+
+.. contents::
+   :local:
+   :depth: 2
+
+
+MCUmgr provides a :file:`mcumgr` command-line tool for managing remote devices.
+The tool is written in the Go programming language.
+
+.. note::
+    This tool is provided for evaluation use only and is not recommended for use in a production environment.
+    It has known issues and will not respect the MCUmgr protocol, for example, upon an error.
+    Instead of aborting the task it will, in some circumstances, sit in an endless loop of sending the same command.
+    Replacement for this tool is currently in development.
+    Once released, support for the Go tool will be dropped.
+    For production environment, use tools listed in the :ref:`zephyr:mcumgr_tools_libraries` section instead.
+
+To install the tool, run the following command:
+
+.. tabs::
+
+   .. group-tab:: go version < 1.18
+
+      .. code-block:: console
+
+         go get github.com/apache/mynewt-mcumgr-cli/mcumgr
+
+   .. group-tab:: go version >= 1.18
+
+      .. code-block:: console
+
+         go install github.com/apache/mynewt-mcumgr-cli/mcumgr@latest
+
+Configuring the transport
+*************************
+
+There are two command-line options that are responsible for setting and configuring the transport layer when communicating with managed device:
+
+* The ``--conntype`` option is used to choose the transport.
+* The ``--connstring`` option is used to pass a comma-separated list of options in the ``key=value`` format, where each valid ``key`` depends on the particular ``conntype``.
+
+Valid transports for ``--conntype`` are ``serial``, ``ble``, and ``udp``.
+Each transport expects a different set of key or value options:
+
+.. tabs::
+
+   .. group-tab:: serial
+
+      ``--connstring`` accepts the following ``key`` values:
+
+      .. list-table::
+         :width: 100%
+         :widths: 10 60
+
+         * - ``dev``
+           - The device name for the OS ``mcumgr`` is running (for example, ``/dev/ttyUSB0``, ``/dev/tty.usbserial``, ``COM1``).
+         * - ``baud``
+           - The communication speed, which must match the baudrate of the server.
+         * - ``mtu``
+           - Maximum Transmission Unit, the maximum protocol packet size.
```

Comment:
Should we mention that the mtu can not be larger then the one set for application? for example, in case of UART the MTU can not be larger than `CONFIG_MCUMGR_TRANSPORT_UART_MTU`.

Reply:
There are also MTU limits for console and UDP, as far as I can tell.

Reply:
There are not, UDP is completely broken in this tool. The text should just be left as-is in my mind